contracts/non-fungible-tokens/andromeda-crowdfund/src/testing/tests.rs

```diff
@@ -1309,6 +1333,206 @@ mod test {
+                            funds: vec![],
+                        }),
+                        ReplyId::PayFee.repr(),
+                    ))),
+            },
+            ClaimTestCase {
+                name: "Claim when cw20 accpeting campaign failed ".to_string(),
+                stage: CampaignStage::FAILED,
+                orders: vec![
+                    TierOrder {
+                        is_presale: true,
+                        amount: Uint128::one(),
+                        level: Uint64::one(),
+                        orderer: orderer.clone(),
+                    },
+                    TierOrder {
+                        is_presale: false,
+                        amount: Uint128::one(),
+                        level: Uint64::one(),
+                        orderer: orderer.clone(),
+                    },
+                ],
+                denom: Asset::Cw20Token(AndrAddr::from_string(MOCK_CW20_CONTRACT.to_string())),
+                expected_res: Ok(Response::new()
+                    .add_attribute("action", "claim")
+                    .add_message(
+                        wasm_execute(
+                            MOCK_CW20_CONTRACT.to_string(),
+                            &Cw20ExecuteMsg::Transfer {
+                                recipient: orderer.to_string(),
+                                amount: Uint128::new(10u128),
+                            },
+                            vec![],
+                        )
+                        .unwrap(),
+                    )
+                    .add_submessage(SubMsg::reply_on_error(
+                        CosmosMsg::Wasm(WasmMsg::Execute {
+                            contract_addr: "economics_contract".to_string(),
+                            msg: to_json_binary(&EconomicsExecuteMsg::PayFee {
+                                payee: orderer.clone(),
+                                action: "Claim".to_string(),
+                            })
+                            .unwrap(),
+                            funds: vec![],
+                        }),
+                        ReplyId::PayFee.repr(),
+                    ))),
+            },
+            ClaimTestCase {
+                name: "Claim without purchasing in successful campaign".to_string(),
+                stage: CampaignStage::SUCCESS,
+                orders: vec![],
+                denom: Asset::NativeToken(MOCK_NATIVE_DENOM.to_string()),
+                expected_res: Err(ContractError::NoPurchases {}),
+            },
+            ClaimTestCase {
+                name: "Claim without purchasing in failed campaign".to_string(),
+                stage: CampaignStage::FAILED,
+                orders: vec![],
+                denom: Asset::NativeToken(MOCK_NATIVE_DENOM.to_string()),
+                expected_res: Err(ContractError::NoPurchases {}),
+            },
+            ClaimTestCase {
+                name: "Claim on invalid stage".to_string(),
```

Comment:
We should add a test that ensures presales do not get a refund on failed campaigns.

Reply:
Claim unit test cases include presale and normal orders. For failed cases, expected refund amount is calculated only for normal orders.